src/wallets/wallets.service.ts

```
@@ -102,38 +102,52 @@ export class WalletsService {
   }
 
   private async getFaucetContractAddress(token: string) {
-    if (token.toLowerCase() === 'mcusd')
-      return {
+    const tokenLowerCase = token.toLowerCase();
+    const contractInfoMap = {
+      mcusd: {
         address: process.env.CELO_CONTRACT_ADDRESS,
         abi: celoFaucetABI,
         provider: this.celoProvider,
         amount: parseUnits('0.15', 18),
         gasPrice: null,
         gasLimit: null,
-      };
-
-    if (token.toLowerCase() === 'apolusdc')
-      return {
+      },
+      apolusdc: {
         address: process.env.MATIC_CONTRACT_ADDRESS,
         abi: maticABI,
         provider: this.maticProvider,
         amount: parseEther('0.5'),
         gasPrice: (await this.maticProvider.getFeeData()).gasPrice,
         gasLimit: 25000,
-      };
+      },
+      usdc: {
+        address: process.env.MATIC_CONTRACT_ADDRESS,
+        abi: maticABI,
+        provider: this.maticProvider,
+        amount: parseEther('0.5'),
+        gasPrice: (await this.maticProvider.getFeeData()).gasPrice,
+        gasLimit: 25000,
+      },
+    };
+
+    return contractInfoMap[tokenLowerCase] || null;
```

I think this is potentially dangerous as null could be returned where a null value wasn't expected we should either throw an error if the value is null or enforce with the types that only the keys defined in the `contractInfoMap` can be passed as function arguments. This can be achieved either with an enum or literal type